### PR TITLE
Structure update

### PR DESCRIPTION
**Pull request by marsicdev**
## Improving project structure
* Project structure reorganized in more flexible way following express4 specifications 
* Mongoose models moved to separate files
* All configuration set in ```config.json```
* Removed unnecessary files, folders and dependencies
* Logging improved 
* Small code refactoring 
* ```README.md``` updated with detailed instructions how to run project
 
Stay tuned more updates and improvements are following...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,80 @@
...
+[httpie](https://github.com/jkbr/httpie) - command line HTTP client
 
 ## Author
 
 This example was created by Evgeny Aleksandrov ([@EAleksandrov](http://twitter.com/EAleksandrov)).
+Updated by [Istock Jared](https://github.com/IstockJared) and  [Marko Arsić](http://portfolio.marsic.info/)  
+
+## License
+
+[MIT](https://github.com/marsicdev/NodeAPI/blob/master/LICENSE)
```
Comment (ealeksandrov) [inline README.md]: Please fix to link original repo.

**Comment by ealeksandrov** [comment]
Thanks! Will look through everything in a couple of days.

File: README.md (hunk 1 above)
Comment (marsicdev) [inline README.md]: Will do :+1:

**Comment by marsicdev** [comment]
Btw great peace of code! I am planing to add more improvements and stuff when I get some free time...

**Comment by ealeksandrov** [comment]
Thanks a lot!

**Pull request merged**